src/web/ViewCubeGizmo.tsx

```diff
@@ -0,0 +1,138 @@
+].map(makePositionVector)
+
+const edgeDimensions = edges.map(
+  (edge) => edge.toArray().map((axis: number): number => (axis == 0 ? 0.5 : 0.25)) as XYZ
+)
+
+const FaceMaterial = ({ hover, index }: { hover: boolean; index: number }) => {
+  const texture = React.useMemo(() => {
+    const canvas = document.createElement('canvas')
+    canvas.width = 128
+    canvas.height = 128
+
+    const context = canvas.getContext('2d')!
+    context.fillStyle = '#eee'
+    context.fillRect(0, 0, canvas.width, canvas.height)
+    context.strokeStyle = '#444'
+    context.strokeRect(0, 0, canvas.width, canvas.height)
+
+    context.font = '28px Arial'
+    context.textAlign = 'center'
+    context.fillStyle = '#222'
+    context.fillText(faces[index].toUpperCase(), 64, 76)
+    return new CanvasTexture(canvas)
+  }, [index])
+  return <meshLambertMaterial map={texture} attachArray="material" color={hover ? 'hotpink' : 'white'} />
+}
+
+const FaceCube = () => {
+  const { tweenCamera, raycast } = useGizmoContext()
+  const [hover, set] = React.useState<number | null>(null)
+  const handlePointerDown = (e: Event) => {
+    tweenCamera(e.face.normal)
+    e.stopPropagation()
+  }
+  const handlePointerMove = (e: Event) => {
+    set(Math.floor(e.faceIndex / 2))
+    e.stopPropagation()
+  }
+  return (
+    <mesh
+      raycast={raycast}
+      onPointerOut={() => set(null)}
+      onPointerMove={handlePointerMove}
+      onPointerDown={handlePointerDown}
+    >
+      {[...Array(6)].map((_, index) => (
+        <FaceMaterial key={index} index={index} hover={hover === index} />
+      ))}
+      <boxBufferGeometry attach="geometry" args={[1, 1, 1]} />
+    </mesh>
+  )
+}
+
+type EdgeCubeProps = {
+  dimensions: XYZ
+  position: Vector3
+}
+
+const EdgeCube = ({ dimensions, position }: EdgeCubeProps): JSX.Element => {
+  const { tweenCamera, raycast } = useGizmoContext()
+  const [hover, set] = React.useState<boolean>(false)
+  const handlePointerOut = (e: Event) => {
+    set(false)
+    e.stopPropagation()
+  }
+  const handlePointerOver = (e: Event) => {
+    set(true)
+    e.stopPropagation()
+  }
+  const handlePointerDown = (e: Event) => {
+    tweenCamera(position)
+    e.stopPropagation()
+  }
+  return (
+    <mesh
+      position={position}
+      raycast={raycast}
+      onPointerOver={handlePointerOver}
+      onPointerOut={handlePointerOut}
+      onPointerDown={handlePointerDown}
+    >
+      <meshBasicMaterial color={hover ? 'hotpink' : 'white'} transparent={true} opacity={0.75} visible={hover} />
+      <boxBufferGeometry attach="geometry" args={dimensions} />
+    </mesh>
+  )
+}
+
+export const ViewCubeGizmo = () => {
+  return (
+    <group scale={[60, 60, 60]}>
+      <FaceCube />
+      {edges.map((edge, index) => (
+        <EdgeCube key={index} position={edge} dimensions={edgeDimensions[index]} />
+      ))}
+      {corners.map((corner, index) => (
+        <EdgeCube key={index} position={corner} dimensions={cornerDimensions} />
```

Comment:
I'm concerned about using `index` as a key. With SSR it's typically fine, but I do like to avoid it. You're mapping vecotrs here, couldn't we use their `uuid` instead?

Reply:
Since the corners and edges are used constants the index as the key should be fine I think. I couldn't find a UUID field in Vector3 (I think you thought of Object3D), but we could use the value of the vector as the key (I added a `keyFor` method, but am happy to revert that).

Reply:
my bad. okay, well lets go with indexes then, and if it causes us any issues we can go a little further into it.